pkg/controller/submariner/submariner_controller.go

```diff
@@ -214,6 +225,33 @@ func (r *ReconcileSubmariner) reconcileDaemonSet(owner metav1.Object, daemonSet
 	return errorutil.WithMessagef(err, "error creating or updating DaemonSet %s/%s", daemonSet.Namespace, daemonSet.Name)
 }
 
+func (r *ReconcileSubmariner)reconcileServiceDiscovery(submariner *submarinerv1alpha1.Submariner, reqLogger logr.Logger, isEnabled bool) error {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		if isEnabled {
+			sd := newServiceDiscoveryCR(submariner)
+			existing, err := r.submClientSet.SubmarinerV1alpha1().ServiceDiscoveries(submariner.Namespace).Get(serviceDiscoveryCrName, metav1.GetOptions{})
+			if err != nil {
+				_, err = r.submClientSet.SubmarinerV1alpha1().ServiceDiscoveries(submariner.Namespace).Create(sd)
+			} else {
+				if reflect.DeepEqual(existing, sd) {
+					// No need to issue the update if the resource didn't change
+					return nil
+				}
+				_, err = r.submClientSet.SubmarinerV1alpha1().ServiceDiscoveries(submariner.Namespace).Update(sd)
+			}
+			if err != nil {
+				reqLogger.Error(err, "Error creating service discovery CR")
+			}
```

Comment:
I think this should use `controllerutil.CreateOrUpdate` with `r.client` here as is done in `reconcileDaemonSet`.   You can then remove `submClientSet`.

Reply:
Removed.

Reply:
I think this should use `controllerutil.CreateOrUpdate` with as is done in `reconcileDaemonSet`.

Reply:
Done